.github/workflows/build.yml

```diff
@@ -283,6 +289,8 @@ jobs:
         with:
           coverage: "none"
           php-version: "7.4"
+          tools: pecl
+          extensions: pthreads
```

Comment:
there's no release for pthreads that supports 7.4, stubs will probably have to be enough :<
I maintain a fork that has superficial 7.4 support, but it's not available via PECL so it's a pain to get hold of

Reply:
I'm trying my luck with `krakjoe/pthreads-polyfill` now.

Reply:
the polyfill is quite old and the API signatures mostly aren't the same as the extension
jetbrains stubs for pthreads should probably be good enough, I updated them recently

Reply:
Yeah but the stubs are not used because somewhere you're extending classes from the extension && use runtime reflection (because classes are autoloaded by Composer) at the same time. I could use `featureToggles.disableRuntimeReflectionProvider` in this test but I want to get closer to your env.

Reply:
Nice, running with `featureToggles.disableRuntimeReflectionProvider: true` actually found a bug :)

Reply:
hmm, that is a problem ... the polyfill isn't going to cut it though, for example [`Thread::start()`](https://github.com/krakjoe/pthreads-polyfill/blob/master/src/Thread.php#L15) has the wrong signature.

You could try using [`pmmp/pthreads:fork74`](https://github.com/pmmp/pthreads/tree/fork74) but it has to be compiled and installed the old-fashioned way without PECL (like this: https://github.com/pmmp/PocketMine-MP/blob/stable/.travis.yml#L9-L17)

We mainly use 7.3 right now because I don't trust my pthreads fork on 7.4 yet (hasn't been tested enough with new features).

Reply:
I'll use `disableRuntimeReflectionProvider: true` instead :)

Reply:
Most of the analysis is the same with it anyway.

Reply:
Yup, probably the best option 👍